### DISCUSSION TITLE
Add `Hishel` into the "Third Party Packages"

### DISCUSSION THREAD

**Discussion by karpetrosyan** [encode/httpx#2797] [Ideas]
Hi folks, I've been working on a project that implements HTTP Caching for `httpx` and `httpcore` projects for the past few weeks, and I'm here to share something interesting with you.

Because it is a **no-cost performance boost**, I think it can be an **excellent httpx companion.**

There is a usage example.

```python
import hishel

cl = hishel.CacheClient()

resp = cl.get('https://www.encode.io')  # takes 0.5222692489624023 seconds
resp = cl.get('https://www.encode.io')  # takes 0.001004934310913086 seconds (520x faster!)
assert resp.extensions['from_cache']
```

It's analogous to [requests-cache](https://github.com/requests-cache/requests-cache), which previously had no acceptable replacements for httpx.

It would be fantastic if you could check out this library ❤️ Together, let's improve the `httpx`.

github: https://github.com/karosis88/hishel
docs: https://hishel.com
pypi: https://pypi.org/project/hishel/

**Comment by lovelydinosaur** [comment]
Ah grand thanks. Yep of course we can add that to the third party packages.
Looks similar to https://github.com/johtso/httpx-caching but... documented.

**Reply by karpetrosyan** [reply]
Although the goal is the same, the design and implementation are completely different. It is expected to clearly implement RFC 9111 while also being very maintainable and reliable.

Thank you very much; I will open a pull request (or issue? I don't think I fully understand when to skip the issues)

**Reply by lovelydinosaur** [reply]
> I will open a pull request (or issue? I don't think I fully understand when to skip the issues)

At a minimum the answer to that would be "if we go through the discussion and decide that it probably doesn't need an issue".

Which is fuzzy but at least we're *defaulting* to:

1. start a discussion
2. we can escalate to an issue if needed
3. resolve the issue with a PR.

I'd suppose that we want issues for anything where we're making a behavioural or tooling change to the project. Minor docs changes it doesn't make as much sense because the change to the docs *is* the description of the issue in itself.
